code/game/objects/items/surgery_tray.dm

```diff
@@ -0,0 +1,338 @@
+			return
+		cautery = arrived
+	else if(istype(arrived, /obj/item/circular_saw))
+		if(circular_saw && circular_saw.toolspeed <= arrived.toolspeed)
+			return
+		circular_saw = arrived
+	else if(istype(arrived, /obj/item/hemostat))
+		if(hemostat && hemostat.toolspeed <= arrived.toolspeed)
+			return
+		hemostat = arrived
+	else if(istype(arrived, /obj/item/surgicaldrill))
+		if(surgical_drill && surgical_drill.toolspeed <= arrived.toolspeed)
+			return
+		surgical_drill = arrived
+	else if(istype(arrived, /obj/item/retractor))
+		if(retractor && retractor.toolspeed <= arrived.toolspeed)
+			return
+		retractor = arrived
+	else if(istype(arrived, /obj/item/scalpel))
+		if(scalpel && scalpel.toolspeed <= arrived.toolspeed)
+			return
+		scalpel = arrived
+	else if(istype(arrived, /obj/item/bonesetter))
+		if(bonesetter)
+			return
+		bonesetter = arrived
+	else if(istype(arrived, /obj/item/razor))
+		if(razor)
+			return
+		razor = arrived
+	else if(istype(arrived, /obj/item/stack/medical/bone_gel))
+		if(bone_gel)
+			return
+		bone_gel = arrived
+	else if(istype(arrived, /obj/item/stack/sticky_tape/surgical))
+		if(surgical_tape)
+			return
+		surgical_tape = arrived
+	else if(istype(arrived, /obj/item/surgical_drapes))
+		if(surgical_drapes)
+			return
+		surgical_drapes = arrived
+
+/obj/item/surgery_tray/Exited(atom/movable/gone, direction)
+	. = ..()
+	if(gone == blood_filter)
+		var/obj/item/blood_filter/other_filter = locate() in contents
+		blood_filter = other_filter || null
+	else if(gone == bonesetter)
+		var/obj/item/bonesetter/other_bonesetter = locate() in contents
+		bonesetter = other_bonesetter || null
+	else if(gone == cautery)
+		var/obj/item/cautery/other_cautery = locate() in contents
+		cautery = other_cautery || null
+	else if(gone == circular_saw)
+		var/obj/item/circular_saw/other_saw = locate() in contents
+		circular_saw = other_saw || null
+	else if(gone == hemostat)
+		var/obj/item/hemostat/other_hemostat = locate() in contents
+		hemostat = other_hemostat || null
+	else if(gone == razor)
+		var/obj/item/razor/other_razor = locate() in contents
+		razor = other_razor || null
+	else if(gone == surgical_drill)
+		var/obj/item/surgicaldrill/other_drill = locate() in contents
+		surgical_drill = other_drill || null
+	else if(gone == retractor)
+		var/obj/item/retractor/other_retractor = locate() in contents
+		retractor = other_retractor || null
+	else if(gone == bone_gel)
+		var/obj/item/stack/medical/bone_gel/other_gel = locate() in contents
+		bone_gel = other_gel || null
+	else if(gone == surgical_tape)
+		var/obj/item/stack/sticky_tape/other_tape = locate() in contents
+		surgical_tape = other_tape || null
+	else if(gone == surgical_drapes)
+		var/obj/item/surgical_drapes/other_drapes = locate() in contents
+		surgical_drapes = other_drapes || null
+
+/obj/item/surgery_tray/proc/PopulateContents()
+	blood_filter = new(src)
+	bonesetter = new(src)
+	razor = new(src)
+	if(!cautery)
```

Comment:
Just for consistency you may as well do the whole `if(!...) x = new(src)` for the other surgical tools. Maybe someone will add an Alien Surgery Tray for Abductors at some point.